src/projectscene/view/clipsview/clipslistmodel.cpp

```diff
@@ -476,43 +496,86 @@ bool ClipsListModel::trimRightClip(const ClipKey& key, bool completed)
 
 void ClipsListModel::selectClip(const ClipKey& key)
 {
-    selectionController()->setSelectedClip(key.key);
-    selectionController()->setSelectedTracks(TrackIdList({ key.key.trackId }));
+    Qt::KeyboardModifiers modifiers = keyboardModifiers();
+
+    if (modifiers.testFlag(Qt::ControlModifier)) {
+        appendToSelectedClips(key);
+    } else {
+        if (muse::contains(selectionController()->selectedClips(), key.key)) {
+            return;
+        }
+        selectionController()->setSelectedClips(ClipKeyList({ key.key }));
+        selectionController()->setSelectedTracks(TrackIdList({ key.key.trackId }));
```

Comment:
This thing used to bother me, now is a good time to discuss it.
I think that the logic of choosing tracks on which the selected clips are located should be in `selectionController`.